### PR TITLE
Add REST APL support

### PR DESCRIPTION
**Pull request by lkostrowski**

### CODE REVIEW

File: env.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
 namespace NodeJS {
   interface ProcessEnv {
-    APL: "file" | "vercel" | "upstash";
+    APL: "file" | "vercel" | "upstash" | "REST";
```
Comment (krzysztofwolski) [inline env.d.ts]:
```suggestion
    APL: "file" | "vercel" | "upstash" | "rest";
```

All names are in the lowercase, lest keep that

File: lib/saleor-app.ts [hunk 2]
```diff
@@ -18,6 +18,15 @@ switch (process.env.APL) {
     // Require `UPSTASH_URL` and `UPSTASH_TOKEN` environment variables
     apl = new UpstashAPL();
     break;
+  case "REST": {
```
Comment (krzysztofwolski) [inline lib/saleor-app.ts]:
```suggestion
  case "rest": {
```

File: lib/saleor-app.ts [hunk 3]
```diff
@@ -18,6 +18,15 @@ switch (process.env.APL) {
...
+    apl = new RestAPL({
+      resourceUrl: process.env.REST_APL_ENDPOINT as string,
+      headers: {
+        Authorization: `Bearer ${process.env.REST_APL_TOKEN as string}`,
```
Comment (krzysztofwolski) [inline lib/saleor-app.ts:29]: Maybe we should log error if the token is not specified?

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: lib/saleor-app.ts (hunk 3 above)
Comment (lkostrowski) [reply lib/saleor-app.ts:29]: done

**Review by krzysztofwolski — APPROVED** [review]

**Pull request merged**